syntax = "proto3";

// TODO(adrian@preemo.io, 02/08/2023): consider other names
enum WorkerStatus {
  WORKER_STATUS_UNSPECIFIED = 0;
  WORKER_STATUS_SUCCESS = 1;
  WORKER_STATUS_FAILURE = 2;
}

message RegisterFunctionReply {
  optional WorkerStatus status = 1;
  optional string message = 2;
}

message WorkerReply {
  oneof reply {
    RegisterFunctionReply register_function = 1;
  }
}
